DOCS-7387 Container Cost Allocation Doc Update (#22176)
DataDog/documentation: content/en/cloud_cost_management/container_cost_allocation.md

Datadog Cloud Cost Management (CCM) automatically allocates the costs of your cloud clusters to individual services and workloads running in those clusters. Use cost metrics enriched with tags from pods, nodes, containers, and tasks to visualize container workload cost in the context of your entire cloud bill.

Clouds
: CCM allocates costs of your AWS, Azure, or Google host instances. A host is a computer (such as an EC2 instance in AWS, a virtual machine in Azure, or a Compute Engine instance in Google Cloud) that is listed in your cloud provider's cost and usage report and may be running Kubernetes pods.

Resources
: CCM allocates costs for Kubernetes clusters and includes cost analysis for many associated resources such as Kubernetes persistent volumes used by your pods. 

CCM displays costs for resources including CPU, memory, and more depending on the cloud and orchestrator you are using on the [**Containers** page][1].

{{< img src="cloud_cost/container_cost_allocation/container_allocation.png" alt="Cloud cost allocation table showing requests and idle costs over the past month on the Containers page" style="width:100%;" >}}

## Prerequisites

{{< tabs >}}
{{% tab "AWS" %}}

> CCM allocates costs of all AWS EC2 instances across your container orchestration environments, including Kubernetes clusters and AWS-specific ECS clusters.

Reviewer: ```suggestion
CCM allocates costs of AWS ECS clusters as well as all Kubernetes clusters, including those managed through Elastic Kubernetes Service (EKS).
```

Making the phrasing here more consistent across the 3 clouds

| Feature | Minimal Agent version | Minimal Cluster Agent version |
|---|---|---|
| Container Cost Allocation | 7.27.0 | 1.11.0 |
| AWS Persistent Volume Allocation | 7.46.0 | 1.11.0  |

1. Configure the AWS Cloud Cost Management integration on the [Cloud Costs Setup page][101].
1. For Kubernetes support, install the [**Datadog Agent**][102] in a Kubernetes environment and ensure that you enable the [**Orchestrator Explorer**][103] in your Agent configuration.
1. For AWS ECS support, set up [**Datadog Container Monitoring**][104] in ECS tasks. 
1. Optionally, enable [AWS Split Cost Allocation][105] for usage-based ECS allocation.

[101]: https://app.datadoghq.com/cost/setup
[102]: /containers/kubernetes/installation/?tab=operator
[103]: /infrastructure/containers/orchestrator_explorer?tab=datadogoperator
[104]: /containers/amazon_ecs/
[105]: https://docs.aws.amazon.com/cur/latest/userguide/enabling-split-cost-allocation-data.html

{{% /tab %}}
{{% tab "Azure" %}}

CCM allocates costs of all Kubernetes clusters, including those managed through Azure Kubernetes Service (AKS). 

1. Configure the Azure Cost Management integration on the [Cloud Costs Setup page][101].

> 1. Install the [**Datadog Agent**][102] in a Kubernetes environment and ensure that you enable the [**Orchestrator Explorer**][103] in your Agent configuration.

Reviewer: Very small nit, for AWS and GCP this comes after the version table below but in Azure it comes before

The following table presents the list of collected features and the minimal Agent and Cluster Agent versions for each.

| Feature | Minimal Agent version | Minimal Cluster Agent version |
|---|---|---|
| Container Cost Allocation | 7.27.0 | 1.11.0 |

[101]: https://app.datadoghq.com/cost/setup
[102]: /containers/kubernetes/installation/?tab=operator
[103]: /infrastructure/containers/orchestrator_explorer?tab=datadogoperator

{{% /tab %}}
{{% tab "Google" %}}

> CCM allocates costs of Google Kubernetes Engine instances to your Google Cloud clusters.

Reviewer: ```suggestion
CCM allocates costs of all Kubernetes clusters, including those managed through Google Kubernetes Engine (GKE).
```

Making this phraising the same as what we have for Azure as I think it is more clear

## Overview

Datadog Cloud Cost Management (CCM) automatically allocates the costs of your cloud clusters to individual services and workloads running in those clusters.

Clouds
: CCM allocates costs of your AWS, Azure, or Google host instances. A host is a computer (such as an EC2 instance in AWS, a virtual machine in Azure, or a Compute Engine instance in Google Cloud) that is listed in your cloud provider's cost and usage report and may be running Kubernetes pods.

Resources
: CCM allocates costs for Kubernetes clusters and includes cost analysis for associated resources such as non-local AWS EBS volumes used by those clusters. 

CCM displays costs for resources including CPU, memory, and more depending on the cloud and orchestrator you are using.

{{< img src="cloud_cost/container_cost_allocation/cost_allocation_table.png" alt="Cloud cost allocation table showing requests and idle costs over the past week" style="width:100%;" >}}

Use cost metrics enriched with tags from your services, nodes, and workloads to visualize individual workload costs in the context of your entire cloud bill. 

## Prerequisites

{{< tabs >}}
{{% tab "Azure" %}}

> CCM allocates costs of Azure VM instances to your Azure Kubernetes Service (AKS) clusters.

Author: ```suggestion
CCM allocates costs of all Kubernetes clusters, including those managed through Azure Kubernetes Service (AKS). 
```

1. Configure the Azure Cost Management integration on the [Cloud Costs Setup page][101].
1. Install the [**Datadog Agent**][102] in a Kubernetes environment and ensure that you enable the [**Orchestrator Explorer**][103] in your Agent configuration.

The following table presents the list of collected features and the minimal Agent and Cluster Agent versions for each.

| Feature | Minimal Agent version | Minimal Cluster Agent version |
|---|---|---|
| Container Cost Allocation | 7.27.0 | 1.11.0 |

[101]: https://app.datadoghq.com/cost/setup
[102]: /containers/kubernetes/installation/?tab=operator
[103]: /infrastructure/containers/orchestrator_explorer?tab=datadogoperator

{{% /tab %}}
{{% tab "AWS" %}}

> CCM allocates costs of AWS EC2 instances to your AWS ECS clusters.

Author: ```suggestion
CCM allocates costs of all AWS EC2 instances across your container orchestration environments, including Kubernetes clusters and AWS-specific ECS clusters.
```

The following table presents the list of collected features and the minimal Agent and Cluster Agent versions for each.

| Feature | Minimal Agent version | Minimal Cluster Agent version |
|---|---|---|
| Container Cost Allocation | 7.27.0 | 1.11.0 |
| AWS Persistent Volume Allocation | 7.46.0 |  |

> | AWS Split Cost Allocation | TBD | |

Author: @elihschiff can you add the minimal Agent version for this feature? 
```suggestion
| AWS Split Cost Allocation | TBD | |
```

Costs are allocated into the following spend types:

| Spend type | Description    |
| -----------| -----------    |
| Managed service fee | Cost of associated fees charged by the provider for managing the cluster, such as Azure fees. |
| Usage | Cost of resources (such as memory and CPU) used by workloads, based on the average usage on that day. |
| Workload idle | Cost of resources (such as memory and CPU) that are reserved and allocated but not used by workloads. This is the difference between the total resources requested and the average usage. |
| Cluster idle | Cost of resources (such as memory and CPU) that are not reserved by workloads in a cluster. This is the difference between the total cost of the resources and what is allocated to workloads. |

{{% /tab %}}
{{% tab "AWS" %}}

### Compute

The cost of a host instance is split into two components: 60% for the CPU and 40% for the memory. Each component is allocated to individual workloads based on their resource reservations and usage.

Costs are allocated into the following spend types:

| Spend type | Description    |
| -----------| -----------    |

> | Managed service fee | Cost of associated fees charged by the provider for managing the cluster, such as AWS ECS fees. |

Reviewer: Managed service fee should probably be in their own section unrelated to compute costs (this goes for all 3 clouds)
I think we should move managed service fees up to a type of resource personally. I would put it as the last resource in the list for each cloud.

Also small typo for each cloud the such as should be: AWS -> EKS, Azure -> AKS, GCP -> GKE

**Note**: Persistent volume allocation is only supported in Kubernetes clusters, and is only available for pods that are part of a Kubernetes StatefulSet.

[101]: https://app.datadoghq.com/integrations/amazon-web-services

{{% /tab %}}
{{% tab "Google" %}}

### Compute

The cost of a host instance is split into two components: 60% for the CPU and 40% for the memory. Each component is allocated to individual workloads based on their resource reservations and usage.

Costs are allocated into the following spend types:

| Spend type | Description    |
| -----------| -----------    |
| Managed service fee | Cost of associated fees charged by the provider for managing the cluster, such as GKE fees. |
| Usage | Cost of resources (such as memory and CPU) used by workloads, based on the average usage on that day. |
| Workload idle | Cost of resources (such as memory and CPU) that are reserved and allocated but not used by workloads. This is the difference between the total resources requested and the average usage. |

> | Cluster idle | Cost of resources (such as memory and CPU) that are not reserved by workloads in a cluster. This is the difference between the total cost of the resources and what is allocated to workloads. |

Reviewer: ```suggestion
| Cluster idle | Cost of resources (such as memory and CPU) that are not reserved by workloads in a cluster. This is the difference between the total cost of the resources and what is allocated to workloads. |
| Not monitored | Cost of resources where the spend type is unknown. Installing the datadog agent on these clusters/nodes will resolve the issue |
```
For GCP only: we need a `Not monitored` section that pretty much tells users that they need to install the datadog agent to get the full support here. I am not sure about my wording though. Feel free to update it

{{% /tab %}}
{{< /tabs >}}

## Understanding resources

Depending on the cloud provider, certain resources may or may not be available for cost allocation.

| Resource | AWS | Azure | Google Cloud |
|---|---|---|---|
| CPU | {{< X >}} | {{< X >}} | {{< X >}} |
| Memory | {{< X >}} | {{< X >}} | {{< X >}} |
| Persistent Volumes | {{< X >}} |  |  |
| Managed service fees | {{< X >}} | {{< X >}} | {{< X >}} |
| ECS costs | {{< X >}} | N/A | N/A |
| Networking costs |  | Limited* | Limited* |
| GPU |  |  | Limited* |
| Local storage |  | Limited* | Limited* |

> For `Limited*` resources, you are required to tag them as part of your Kubernetes cost allocation using the `allocated_spend_type:<resource>_not_supported` tagging format, for example: `allocated_spend_type:gpu_not_supported`. These resources are host-level costs, not pod or namespace-level costs.

Reviewer: ```suggestion
`Limited*` resources have been identified to be a part of your Kubernetes spend, but are not currently fully allocated to specific workloads or pods. These resources are host-level costs, not pod or namespace-level costs. You will see `allocated_spend_type:<resource>_not_supported` on these resources. 
```

> 

Reviewer: In the above table for `Out-of-the-box tag` I think we should rework it to just be a table that describes the `orchestrator` tag. (dropping all the other rows as they are becoming redundant with the rest of the page)

The orchestrator section should say something about how we add a `orchestrator` tag to all resources associated with your cluster. 

If there could be a tab per cloud here as well. Azure and GCP will have the table with a single row for `orchestrator:kubernetes` while AWS will also have the `orchestrator:ecs` row
